gradle-failure-reports/src/main/java/com/palantir/gradle/failurereports/BuildFailureReporter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.failurereports;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.palantir.gradle.failurereports.junit.JunitReporter;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.gradle.api.Task;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.plugins.quality.Checkstyle;
+import org.gradle.api.tasks.TaskExecutionException;
+import org.gradle.api.tasks.compile.JavaCompile;
+import org.gradle.execution.MultipleBuildFailures;
+
+public final class BuildFailureReporter {
+
+    private static Logger log = Logging.getLogger(BuildFailureReporter.class);
+
+    public static void report(File outputFile, Throwable buildThrowable) {
+        Optional.ofNullable(buildThrowable).ifPresent(failure -> {
+            try {
+                reportFailures(outputFile, failure);
+            } catch (IOException e) {
+                log.error("Failed to report build failures", e);
+            }
+        });
+    }
+
+    private static void reportFailures(File outputFile, Throwable buildThrowable) throws IOException {
+        ImmutableList.Builder<Throwable> rootExceptions = ImmutableList.builder();
+        ImmutableList.Builder<FailureReport> failureReports = ImmutableList.builder();
+        if (buildThrowable instanceof MultipleBuildFailures) {
+            rootExceptions.addAll(((MultipleBuildFailures) buildThrowable).getCauses());
+        } else {
+            rootExceptions.add(buildThrowable);
+        }
+        List<TaskExecutionException> throwables = rootExceptions.build().stream()
+                .map(Throwables::getCausalChain)
+                .flatMap(Collection::stream)
+                .filter(throwable -> throwable instanceof TaskExecutionException)
```

Comment:
- TaskExecutionExceptions are wrapped into a LocationAware exception 
- I think with this workflow we can report all the errors that come not only from tasks